spannerlib/wrappers/spannerlib-python/spannerlib-python/google/cloud/spannerlib/connection.py

```diff
@@ -69,3 +79,108 @@ def _close_lib_object(self) -> None:
                 "Unexpected error closing connection ID: %d", self.oid
             )
             raise SpannerLibError(f"Unexpected error during close: {e}") from e
+
+    def execute(self, request: ExecuteSqlRequest) -> Rows:
+        """Executes a SQL statement on the connection.
+
+        Args:
+            request (ExecuteSqlRequest): The ExecuteSqlRequest object.
+
+        Returns:
+            A Rows object representing the result of the execution.
+        """
+        if self.closed:
+            raise RuntimeError("Connection is closed.")
+
+        logger.info(
```

Comment:
Here also: Consider adjusting the log level. The log of an application should not be spammed with a lot of 'unnecessary' information from a library like this, unless it has been explicitly configured.